1109-i386-txt-Add-Intel-TXT-definitions-header-file.patch

```diff
@@ -0,0 +1,712 @@
++}
++
++static inline struct grub_txt_sinit_mle_data*
++grub_txt_sinit_mle_data_start (grub_uint8_t *heap)
++{
++  return (struct grub_txt_sinit_mle_data*)(heap +
++                 grub_txt_bios_data_size (heap) +
++                 grub_txt_os_mle_data_size (heap) +
++                 grub_txt_os_sinit_data_size (heap) +
++                 sizeof (grub_uint64_t));
++}
++
++/* Intel 64 and IA-32 Architectures Software Developer’s Manual */
++/* Volume 2 (2A, 2B, 2C & 2D): Instruction Set Reference, A-Z */
++
++/* CHAPTER 6 SAFER MODE EXTENSIONS REFERENCE */
++
++#define GRUB_SMX_LEAF_CAPABILITIES	0
++#define GRUB_SMX_LEAF_UNDEFINED		1
++#define GRUB_SMX_LEAF_ENTERACCS		2
++#define GRUB_SMX_LEAF_EXITAC 		3
++#define GRUB_SMX_LEAF_SENTER		4
++#define GRUB_SMX_LEAF_SEXIT		5
++#define GRUB_SMX_LEAF_PARAMETERS	6
++#define GRUB_SMX_LEAF_SMCTRL		7
++#define GRUB_SMX_LEAF_WAKEUP		8
++
++#define GRUB_SMX_CAPABILITY_CHIPSET_PRESENT	(1<<0)
++#define GRUB_SMX_CAPABILITY_UNDEFINED		(1<<1)
++#define GRUB_SMX_CAPABILITY_ENTERACCS		(1<<2)
++#define GRUB_SMX_CAPABILITY_EXITAC		(1<<3)
++#define GRUB_SMX_CAPABILITY_SENTER		(1<<4)
++#define GRUB_SMX_CAPABILITY_SEXIT		(1<<5)
++#define GRUB_SMX_CAPABILITY_PARAMETERS		(1<<6)
++#define GRUB_SMX_CAPABILITY_SMCTRL		(1<<7)
++#define GRUB_SMX_CAPABILITY_WAKEUP		(1<<8)
++#define GRUB_SMX_CAPABILITY_EXTENDED_LEAFS	(1<<31)
++
++static inline grub_uint32_t
++grub_txt_getsec_capabilities (grub_uint32_t index)
++{
++  grub_uint32_t caps;
++
++  asm volatile (".byte 0x0f,0x37\n"
++                        : "=a" (caps)
++                        : "a" (GRUB_SMX_LEAF_CAPABILITIES), "b" (index));
++  return caps;
++}
++
++static inline void
++grub_txt_getsec_enteraccs (grub_uint32_t acm_phys_addr, grub_uint32_t acm_size)
++{
++  asm volatile (".byte 0x0f,0x37\n" :
++                        : "a" (GRUB_SMX_LEAF_ENTERACCS),
++                          "b" (acm_phys_addr), "c" (acm_size));
++}
++
++static inline void
++grub_txt_getsec_exitac (grub_uint32_t near_jump)
++{
++  asm volatile (".byte 0x0f,0x37\n" :
++                        : "a" (GRUB_SMX_LEAF_EXITAC), "b" (near_jump));
++}
++
++static inline void
++grub_txt_getsec_senter (grub_uint32_t acm_phys_addr, grub_uint32_t acm_size)
++{
++  asm volatile (".byte 0x0f,0x37\n" :
++                        : "a" (GRUB_SMX_LEAF_SENTER),
++                          "b" (acm_phys_addr), "c" (acm_size));
++}
++
++static inline void
++grub_txt_getsec_sexit (void)
++{
++  asm volatile (".byte 0x0f,0x37\n" : : "a" (GRUB_SMX_LEAF_SEXIT));
++}
++
++#define GRUB_SMX_PARAMETER_TYPE_MASK		0x1f
++#define GRUB_SMX_PARAMETER_NULL			0
++#define GRUB_SMX_PARAMETER_ACM_VERSIONS		1
++#define GRUB_SMX_PARAMETER_MAX_ACM_SIZE		2
++#define GRUB_SMX_PARAMETER_ACM_MEMORY_TYPES	3
++#define GRUB_SMX_PARAMETER_SENTER_CONTROLS	4
++#define GRUB_SMX_PARAMETER_TXT_EXTENSIONS	5
++
++
++#define GRUB_SMX_PARAMETER_MAX_VERSIONS	0x20
++
++#define GRUB_SMX_GET_MAX_ACM_SIZE(v)	((v & ~GRUB_SMX_PARAMETER_TYPE_MASK)*0x20)
```

Comment:
```suggestion
+#define GRUB_SMX_GET_MAX_ACM_SIZE(v)	((v) & ~(__typeof__(v))GRUB_SMX_PARAMETER_TYPE_MASK)
```

My reading of the spec is that bits 31:5 hold the value that needs to be multiplied by 32, but it must first be shifted right 5 bits, which is equivalent to dividing by 32.  Therefore, the shift and multiply cancel.  The cast is to prevent truncation.